sycl/include/CL/sycl/ONEAPI/reduction.hpp

```diff
@@ -30,6 +31,15 @@ using cl::sycl::detail::is_sgeninteger;
 using cl::sycl::detail::queue_impl;
 using cl::sycl::detail::remove_AS;
 
+// std::tuple seems to be a) too heavy and b) not copyable to device now
```

Comment:
Could you please clarify what specifically makes `std::typle` heavy?

Reply:
I'll eventually have to do that. At this moment I only know that SPIR-V file for a reduction test with std::tuple is 34% bigger than SPIRV for the same reduction test with sycl::detail::tuple (i.e. with this change).